core/src/wsv.rs

```diff
@@ -342,15 +474,14 @@ impl WorldStateView {
     ///
     /// # Errors
     /// Fails if transaction instruction execution fails
-    async fn execute_transactions(&self, block: &CommittedBlock) -> Result<()> {
+    fn execute_transactions(&mut self, block: &CommittedBlock) -> Result<()> {
         // TODO: Should this block panic instead?
         for tx in &block.transactions {
             self.process_executable(
                 &tx.as_v1().payload.instructions,
                 tx.payload().account_id.clone(),
             )?;
             self.transactions.insert(tx.hash());
-            task::yield_now().await;
```

Comment:
Good catch. I would personally be too afraid to check if this is actually useful.